app/tree/gui/src/ts/twxplore/gui/tree/components/map/Map.tsx

```diff
@@ -2,128 +2,243 @@ import {connect, useDispatch, useSelector} from "react-redux";
-    const dispatch = useDispatch();
-    const state: MapState = useSelector((rootState: RootState) => rootState.app.map);
-    // Load boroughs on first render
-    const boroughsQueryResult = useQuery<BoroughsQuery>(boroughsQueryDocument, {});
-    
-    const [getNtasByBorough,{}] = useLazyQuery<NtasByBoroughQuery, NtasByBoroughQueryVariables>(ntasByBoroughQueryDocument, {
-        onCompleted: (data: NtasByBoroughQuery) => {
-            dispatch(addMapFeatures(data.ntas.byBoroughGeometry.map(ntaFeature => ({
-            childType: MapFeatureType.BLOCKFACE,
+  const dispatch = useDispatch();
+  const state: MapState = useSelector(
+    (rootState: RootState) => rootState.app.map
+  );
+  // Load boroughs on first render
+  const boroughsQueryResult = useQuery<BoroughsQuery>(
+    boroughsQueryDocument,
+    {}
+  );
+
+  /* Lazy Query to get Ntas that belong to a borough.
+  The onCompleted function dispatches an action that
+  puts each nta from the query into the feature list
+  on the state with a feature-state of 'LOADED'
+  */
+  const [getNtasByBorough, {}] = useLazyQuery<
+    NtasByBoroughQuery,
+    NtasByBoroughQueryVariables
+  >(ntasByBoroughQueryDocument, {
+    onCompleted: (data: NtasByBoroughQuery) => {
+      dispatch(
+        addMapFeatures(
+          data.ntas.byBoroughGeometry.map(ntaFeature => ({
+            childType: MapFeatureType.BLOCK,
             geometry: ntaFeature.geometry,
             state: MapFeatureState.LOADED,
             type: MapFeatureType.NTA,
-            uri: ntaFeature.uri
-        }))))}
-        
-    })
+            uri: ntaFeature.uri,
+          }))
+        )
+      );
+    },
+  });
 
-    if (state.features.length === 0) {
-        if (boroughsQueryResult.data) {
-            // No tracking any features yet, add the boroughs we loaded
-            dispatch(addMapFeatures(boroughsQueryResult.data.boroughs.geometries.map(boroughFeature => ({
-                childType: MapFeatureType.NTA,
-                geometry: boroughFeature.geometry,
-                state: MapFeatureState.LOADED,
-                type: MapFeatureType.BOROUGH,
-                uri: boroughFeature.uri
-            }))));
-        }
+  /* Same lazy query above except to get Blocks under an Nta*/
+  const [getBlocksByNta] = useLazyQuery<
+    BlocksByNtaQuery,
+    BlocksByNtaQueryVariables
+  >(blocksbyNtaQueryDocument, {
+    onCompleted: (data: BlocksByNtaQuery) => {
+      dispatch(
+        addMapFeatures(
+          data.blocks.byNtaGeometry.map(blockFeature => ({
+            childType: MapFeatureType.TREE,
+            geometry: blockFeature.geometry,
+            state: MapFeatureState.LOADED,
+            type: MapFeatureType.BLOCK,
+            uri: blockFeature.uri,
+          }))
+        )
+      );
+    },
+  });
+
+  /* This is a very comprehensive query. Takes in a filter that will specifify what info to return*/
+  const [getTreeMap, {}] = useLazyQuery<TreeMapQuery, TreeMapQueryVariables>(
```

Comment:
This query name is a misnomer, please rename. It should be TreesBySelectionQuery or similar.